src/actions/app.ts

```diff
@@ -8,13 +8,25 @@ Code distributed by Google as part of the polymer project is also
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 */
 
+import { Action, ActionCreator } from 'redux';
+import { ThunkAction } from 'redux-thunk';
+import { RootState } from '../store.js';
 export const UPDATE_PAGE = 'UPDATE_PAGE';
 export const UPDATE_OFFLINE = 'UPDATE_OFFLINE';
 export const UPDATE_DRAWER_STATE = 'UPDATE_DRAWER_STATE';
 export const OPEN_SNACKBAR = 'OPEN_SNACKBAR';
 export const CLOSE_SNACKBAR = 'CLOSE_SNACKBAR';
 
-export const navigate = (path) => (dispatch) => {
+export interface AppActionUpdatePage extends Action<'UPDATE_PAGE'> {page: string};
```

Comment:
@keanulee why export these interfaces?

Reply:
Oh, I guess store.js only uses `AppAction`, not these interfaces. No good reason, but maybe someone might want to create the action before dispatching it?

```js
const action = actionCreator(param);
store.dispatch(action);
```